File: CustomerSales/src/Entities/SalesOpportunity.cs
namespace Api.Entities
{
    public record SalesOpportunity
    {
        private SalesOpportunityStatusEnum _status;
        
        public Guid SalesOpportunityId { get; set; }

        public string Status
        {
            get => _status.ToString();
            set => _status = (SalesOpportunityStatusEnum)Enum.Parse(typeof(SalesOpportunityStatusEnum), value);
        }
        public string Name { get; set; }

        public SalesOpportunity()
        {
            SalesOpportunityId = Guid.NewGuid();
            _status = SalesOpportunityStatusEnum.Unknown;
            Name = string.Empty;
        }
    }
}

File: ui/src/Entities/SalesOpportunity.ts
export default class SalesOpportunity {
    public salesOpportunityId: string
    public status: string
    public name: string

    constructor(id: string, status: string, name: string) {
        this.salesOpportunityId = id
        this.status = status
        this.name = name
    }
}
